activesupport/test/cache/behaviors/local_cache_behavior.rb

```diff
@@ -19,21 +20,26 @@ def test_instrumentation_with_local_cache
   end
 
   def test_local_writes_are_persistent_on_the_remote_cache
+    key = SecureRandom.uuid
+    value = SecureRandom.alphanumeric
     retval = @cache.with_local_cache do
-      @cache.write("foo", "bar")
+      @cache.write(key, value)
     end
     assert retval
-    assert_equal "bar", @cache.read("foo")
+    assert_equal value, @cache.read(key)
   end
 
   def test_clear_also_clears_local_cache
+    skip "MemCached Store is failing on this test..."
```

Comment:
This spec is failing only on `MemCached`, in my understanding this should not happen. I'm not sure how it used to be OK before the change.

![image](https://user-images.githubusercontent.com/1037088/143333460-79c8f4ac-52b1-4aa0-977e-4b2de63bbea1.png)

Reply:
I think we found a bug in the local cache. Maybe what is happening is that the local cache is not being cleaned?

Reply:
yeah, maybe it's a bug. I'll investigate more on this and be back to you with my findings...

Reply:
After a few I was able to locate the reason it's only failing on `MemCached` store:

https://github.com/rails/rails/blob/2a32c4b679a7fdc370d2f635c5285e4a4f161390/activesupport/test/cache/stores/mem_cache_store_test.rb#L78-L87

Turns out this particular test (`test_clear_also_clears_local_cache`) is override after loaded, and [since we can't actually use `@cache.clear`](https://github.com/rails/rails/blob/2a32c4b679a7fdc370d2f635c5285e4a4f161390/activesupport/lib/active_support/cache/mem_cache_store.rb#L156-L162) without disrupting all other test it was changed to not use `flush_all` and just delete the current key composed by  _namespace_ plus `foo`, when changed using a random key, it started to fail.

So it's not a bug on the local cache nor the `clear` method.

Reply:
I've added https://github.com/rails/rails/pull/43718/commits/c677229b2c6dbeff1aeef0f14865dfb8b9153e35, to fix this problem with `MemCached` store... it this approach acceptable, @rafaelfranca ?